### ISSUE TITLE
Fine-tune cquery support

### ISSUE THREAD

**Issue by joaotavora** [joaotavora/eglot#9]
@jaelsasser here's a fresh issue to continue the discussion started in #6.

Regarding the renaming issue, it's of course nicer to implement some kind of `$fileDump` request, but I wonder if it isn't simpler just to put some kind of printf somewhere in `cquery`. 

~That is, I think it might very well be eglot who is not reporting the changes correctly. Other servers are probably more lenient on this, but I more or less recall the spec saying something about overlapping changes and how you shouldn't do that. Well, Eglot does it.~

~I don't think it's a question of reporting them twice, because other servers would have suffered from the same error.~

EDIT: read below, I ruled out the above hypotheses. it's on cquery, I think.

Another problem I'm having is that [I can't get my git-built cquery to load my simple project](https://github.com/cquery-project/cquery/issues/691). ~Is there a simple project guaranteed to work on a brand-new built cquery?~ Actually, never mind, I just got a minuscule `hello-world.c` to load.

**Comment by joaotavora** [comment]
@jaelsasser do I did the printf trick and it seems the problem in on cquery's side. This was my hack:
```diff
diff --git a/src/working_files.cc b/src/working_files.cc
index 83fa7eab..39b91ad5 100644
--- a/src/working_files.cc
+++ b/src/working_files.cc
@@ -524,6 +524,8 @@ void WorkingFiles::OnChange(const lsTextDocumentDidChangeParams& change) {
       file->buffer_content.replace(file->buffer_content.begin() + start_offset,
                                    file->buffer_content.begin() + end_offset,
                                    diff.text);
+      LOG_S(WARNING) << "Here are some contents:\nSTART\n " << file->buffer_content
+                     << "\nEND";
       file->OnBufferContentUpdated();
     }
   }
```
Horribly, but usefully, this prints out the file contents to stderr for every incremental change eglot sends. There is never a discrepancy in the contents themselves, but indeed the highlighting gets very confused, even with `:enableIndexOnChange t`

**Comment by jaelsasser** [comment]
Well, that's both comforting that its not in eglot and a pain that its probably somewhere deep in the bowels of cquery. I can dig around and refamiliarize myself with the diagnostics and highlighting to try and find out where this is falling apart.

I've also got one thing to add to the scope of this issue: some `std` C++ functions with overloaded arguments start to get a little ridiculous with multiple signatures. I've got a local patch locally that turns the `cl-loop` in `eglot--sig-info` to an `until` rather than an accumulator, which makes the mini buffer a lot more readable.
